dsdl_compiler/libcanard_dsdl_compiler/code_type_template.tmpl

```diff
@@ -223,11 +230,16 @@ int32_t ${type_name}_decode_internal(
     for (c = 0; c < dest->${'%s' % ((f.name + '.len'))}; c++)
     {
                     %if isinstance(f.data_type.element_type, CompoundType)
-        offset += ${f.c_type}_decode_internal(transfer,
+        offset = ${f.c_type}_decode_internal(transfer,
```

Comment:
The way we're treating `offset` here seems counterintuitive.  It would appear that the value should be advanced by the returned number of bits.